### PR TITLE
Remove obsolete system settings

### PR DESCRIPTION
**Pull request by alroniks**
### What does it do?
Removes obsolete system settings.

### Why is it needed?
They not used in the system anymore and are messing up the code.

### Related issue(s)/PR(s)
#13858

### CODE REVIEW

**Comment by Mark-H** [comment]
They should probably also be removed in an upgrade script in the setup - just removing them from the build is not enough to make them actually go away.

**Review by bezumkin — CHANGES_REQUESTED** [review]
I think we also need to delete these settings when updating the site.

**Comment by alroniks** [comment]
Good catch @Mark-H. I will update the pr.

**Comment by alroniks** [comment]
@bezumkin @Mark-H PR is updated. Please, have a look.

**Review by bezumkin — DISMISSED** [review]

**Review by opengeek — DISMISSED** [review]

**Comment by alroniks** [comment]
@opengeek @bezumkin Excuse me for troubling but I updated upgrading script to avoid silence mode. Now it says what was removed and errors in the case when system settings cannot be removed.

File: setup/includes/upgrades/common/3.0.0-cleanup-system-settings.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+foreach ($settings as $key) {
+    /** @var modSystemSetting $setting */
+    $setting = $modx->getObject('modSystemSetting', ['key' => $key]);
+    if ($setting instanceof modSystemSetting) {
+        if ($setting->remove()) {
+            $this->runner->addResult(modInstallRunner::RESULT_SUCCESS,
+                sprintf($messageTemplate, 'ok', $this->install->lexicon('system_setting_cleanup_success', ['key' => $key])));
+        } else {
+            $this->runner->addResult(modInstallRunner::RESULT_FAILURE,
```
Comment (opengeek) [inline setup/includes/upgrades/common/3.0.0-cleanup-system-settings.php]: I don't think we want this be a failure if it doesn't remove a system setting. Do we?

Comment (opengeek) [reply setup/includes/upgrades/common/3.0.0-cleanup-system-settings.php]: I think warning there, and nothing if the setting doesn't exist. A warning if it has already been removed is unnecessary in my opinion.

Comment (alroniks) [reply setup/includes/upgrades/common/3.0.0-cleanup-system-settings.php]: Yes, make sense. I updated this part.

**Review by gpsietzema — COMMENTED** [review]
I can't seem to get this to work on PHP7.2/Mariadb 10.2.14 (mysql 5.5.5).

I do get the message in the setup that the 2 settings have been removed, but they are still there.

Another thing: shouldn't compress_js also be removed. And what about compress_css, we are keeping that one I think?

**Comment by alroniks** [comment]
@gpsietzema Did you try to rebuild core package after checkout? Because it removes settings during the upgrade but if core package is not rebuilt it sets up these settings from the package again.

**Review by gpsietzema — APPROVED** [review]
@Alroniks You are right, rebuilding the core did the job!

Still wondering about compress_js settings. Shouldn't it be removed as well?

**Comment by bezumkin** [comment]
@gpsietzema `compress_js` is still in use, because we have [minified version](https://github.com/modxcms/revolution/blob/2.x/manager/assets/modext/modx.jsgrps-min.js)

**Comment by gpsietzema** [comment]
@bezumkin but it was removed from the installer somehow... or is that totally unrelated?

**Comment by bezumkin** [comment]
@gpsietzema https://github.com/modxcms/revolution/pull/13785#issuecomment-380115763 - looks like we use minified versions by default, but can disable it for development.

So, this settings don`t need in setup.

**Review by Mark-H — APPROVED** [review]

**Pull request merged**